File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-0.go
func minSubArrayLen(target int, nums []int) int {

	l := 0
	res := len(nums)+1
	sum := 0

	for r := 0; r < len(nums); r++ {

		sum += nums[r]

		for sum >= target {

			res = min(res, r - l+1)

			sum -= nums[l]
			l++
		}


	}

	if res != len(nums) + 1 {
		return res
	} else {
		return 0
	}

}

File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        n = len(nums)

        res, subset = [], []

        def dfs(i):
            if i >= len(nums):
                if sum(subset) == target:
                    res.append(subset.copy())
                return
            
            if sum(subset) > target:
                return

            # Include nums[i]
            subset.append(nums[i])
            dfs(i)
            subset.pop()
            dfs(i+1)

        dfs(0)
        return res  
        